simtools/simtel/simtel_config_writer.py

```diff
@@ -288,9 +299,104 @@ def _write_site_parameters(self, file, site_model):
                 search_telescope_parameters=False,
                 search_site_parameters=True,
             )
+            _simtel_name, value = self._convert_model_parameters_to_simtel_format(
+                _simtel_name, value, model_path, telescope_model
+            )
             if _simtel_name is not None:
                 file.write(f"{self.TAB}{_simtel_name} = {value}\n")
         _simtel_meta = self._get_simtel_metadata("site")
         for _simtel_name, value in _simtel_meta.items():
             file.write(f"{self.TAB}{_simtel_name} = {value}\n")
         file.write("\n")
+
+    def _convert_model_parameters_to_simtel_format(
+        self, simtel_name, value, model_path, telescope_model
+    ):
+        """
+        Convert model parameter value to simtel format.
+
+        This might involve format or unit conversion and writing to a parameter file.
+
+        Parameters
+        ----------
+        simtel_name: str
+            Parameter name.
+        value: any
+            Value to convert.
+        model_path: Path
+            Path to the model for writing of additional files.
+        telescope_model: dict of TelescopeModel
+            Telescope models.
+
+        Returns
+        -------
+        str, any
+            Converted parameter name and value.
+        """
+        if simtel_name == "array_triggers":
+            value = self._write_array_triggers_file(value, model_path, telescope_model)
+        return simtel_name, value
+
+    def _write_array_triggers_file(self, array_triggers, model_path, telescope_model):
+        """
+        Write array trigger definition file in simtel format.
+
+        Parameters
+        ----------
+        array_triggers: dict
+            Array trigger definitions.
+        model_path: Path
+            Path to the model for writing of additional files.
+        telescope_model: dict of TelescopeModel
+            Telescope models.
+        """
+        trigger_per_telescope_type = {}
+        for count, tel_name in enumerate(telescope_model.keys()):
+            telescope_type = names.get_array_element_type_from_name(tel_name)
+            trigger_per_telescope_type.setdefault(telescope_type, []).append(count + 1)
+
+        trigger_lines = {}
+        for tel_type, tel_list in trigger_per_telescope_type.items():
+            trigger_dict = self._get_array_triggers_for_telescope_type(array_triggers, tel_type)
+            trigger_lines[tel_type] = f"Trigger {trigger_dict['multiplicity']['value']} of "
+            trigger_lines[tel_type] += ", ".join(map(str, tel_list))
+            if all(trigger_dict["width"][key] is not None for key in ["value", "unit"]):
```

Comment:
I shortened these `if` statements a bit, but what worries me is the case in which they are `None`. We do not raise an error and instead rely on sim_telarray to fail (which will be hard to find). Should we raise an error here?

Reply:
see issue #1200